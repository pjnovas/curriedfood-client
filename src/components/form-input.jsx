import React from 'react';
import { StyleSheet } from 'react-native';
import { TextInput, Caption } from 'react-native-paper';
import { useFormikContext } from 'formik';
import Theme from '../theme';

// TODO: Change Caption into HelperText for showing errors
// https://callstack.github.io/react-native-paper/helper-text.html

export const FormInput = ({ id, ...inputProps }) => {
  const formContext = useFormikContext();
  const { [id]: error } = formContext.errors;

  return (
    <>
      <TextInput
        {...inputProps}
        error={!!error}
        onChangeText={formContext.handleChange(id)}
        mode="outlined"
      />
      {error && <Caption style={styles.error}>{error}</Caption>}
    </>
  );
};

const styles = StyleSheet.create({
  error: {
    color: Theme.colors.error
  }
});
