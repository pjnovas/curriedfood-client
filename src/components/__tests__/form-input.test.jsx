import { FormInput } from '../form-input.jsx';

jest.mock('formik', () => ({
  useFormikContext: () => ({
    handleChange: jest.fn(),
    errors: {
      'input-id': 'BOOM'
    }
  })
}));

takeSnapshots(
  {
    default: {
      id: 'input-id',
      type: 'text'
    }
  },
  FormInput
);
