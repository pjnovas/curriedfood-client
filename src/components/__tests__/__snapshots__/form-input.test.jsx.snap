// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
<Fragment>
  <withTheme(TextInput)
    error={true}
    mode="outlined"
    type="text"
  />
  <Caption
    style={
      Object {
        "color": "#CF6679",
      }
    }
  >
    BOOM
  </Caption>
</Fragment>
`;
